src/command/dns-command.ts

```diff
@@ -165,6 +165,7 @@ export class DnsCommand implements CommandInterface<DnsOptions> {
 			result = {
 				status: 'failed',
 				rawOutput: 'Private IP ranges are not allowed',
+				...(!options.trace ? { resolver: (result as DnsParseResponseClassic).resolver } : {}),
```

Comment:
Why not trace? 🤔 It seems the field is there for non-error cases.

Reply:
for trace resolver is stored inside hops objects